oauth2-example-app/src/main/java/com/clouway/oauth2/exampleapp/storage/InMemoryUserRepository.java

```diff
@@ -27,7 +28,7 @@
   }
 
   @Override
-  public Optional<Identity> findIdentity(String identityId, DateTime instantTime) {
+  public Optional<Identity> findIdentity(String identityId, GrantType grantType, DateTime instantTime) {
```

Comment:
Here at the implementation in the example app we could make a simple usage of the grantType with the Identity just to show its purpose and make the idea clearer.